WordPress/wordpress-playground
# Blueprints: Add resetData step

Adds a resetData step that deletes WordPress posts and comments and sets the auto increment sequence for the
posts and comments tables to 0.

Blueprint developers often want to use the last post ID in `landingPage`. However, different WordPress releases
provide a different number of posts, which means a different starting
ID. When you create a post, it may have an ID=4 or ID=6 depending on the
WordPress release.

The `resetData` step solves that problem by ensuring WordPress posts
start at ID=1. Importing or creating 5 posts will reliably yield five
posts with IDs from 1 to 5.

The step is marked as private and internal for now so we can confirm
it's actually useful and needed before advertising it as public API.

 ## Example Usage

```json
{
  "steps": [
    { "step": "resetData" }
  ],
  "login": true,
  "landingPage": "/wp-admin/edit.php"
}
```

 ## Alternatives considered

The WXR importer used in Playground supports specifying `post_id` in WXR. You could bring in WXR with post id=1, 2, all the way to 10, and then you’d be able to use ?p=1 or ?p=10 as a landing page. Caveat: You’ll need to clear the wp_posts table before that. We don’t know what would happen when you don’t.

 ## Testing instructions

Confirm the CI checks pass. Also, try the usage example above, create a
new post, and confirm it has ID=1.
